### ISSUE TITLE
JSONPath attribute matching

### ISSUE THREAD

**Issue by dsander** [cantino/huginn#836]
I am trying to select objects out of an array using JSONPath based on on an attribute matching a specific value.

The WebsiteAgent is fetching https://ws.ovh.com/dedicated/r2/ws.dispatcher/getAvailability2 and 
the JSONPath I am using is `$.answer.availability[?(@.reference='150sk30')].metaZones..availability`

I would expect an output like this:

```
[  
   "unknown",
   "unknown",
   "unknown",
   "unknown",
   "unknown"
]
```

but it return an array containing all availabilities not matter if the reference matches or not.

To me it looks like a limitation of the JSONPath ruby library we are using. 

Is there a way around it without passing the JSON to a JavascriptAgent extracting the values 'manually'?

**Comment by cantino** [comment]
Unfortunately, I don't know JSONPath well enough to really answer the question.  (The JavaScriptAgent could definitely do this, of course.)  We could fork and extend the JSONPath library if it's limiting.

**Comment by dsander** [comment]
This was the first time I tried to use it :smile: Actually I thought the JSONPath library could use a rewrite, having to enable `eval` to allow the attribute matching felt wrong to me.

The JSONPath ['specs'](http://goessner.net/articles/JsonPath/) don't look to hard to parse without using eval.

**Comment by virtadpt** [comment]
Would the JSONpath Online Evaluator (http://ashphy.com/JSONPathOnlineEvaluator/) help?

**Comment by dsander** [comment]
@virtadpt I used a different online tool to validate my JSONPath expression and it really seems the library we use does not support attribute matching.

I am still undecided if I am motivated enough to rewrite it, removing the need to `eval` would be pretty great.

**Comment by virtadpt** [comment]
Which tool do you use, if you don't mind my asking?
Are there viable alternatives out there to the library used right now?

**Comment by dsander** [comment]
@virtadpt I used http://jsonpath.curiousconcept.com/

For ruby the library we use seems to be the only one available.

**Comment by dsander** [comment]
Closing this since I solved my initial problem by using the JavascriptAgent.

**Comment by virtadpt** [comment]
Oh, the one at Curiousconcept.  I use that one occasionally.

As for the Ruby gem... I can do some asking around and see if my cow-orkers know of a different one, but that may not be the case.
